third_party/terraform/utils/provider_test.go.erb

```diff
@@ -96,6 +106,88 @@ func init() {
 	<% end -%>
 }
 
+var lock = &sync.Mutex{}
+
+// Returns a cached config if VCR testing is enabled. This enables us to use a single HTTP transport
+// for a given test, allowing for recording of HTTP interactions.
+func getCachedConfig(d *schema.ResourceData, configureFunc func(d *schema.ResourceData) (interface{}, error), testName string) (interface{}, error) {
```

Comment:
Since we know this always returns a Config, could it return that instead of a interface{}?